packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -247,6 +269,110 @@ class ImageCache {
   }
 }
 
+/// A class that provides information about how the [FlutterImageCache] is
+/// tracking an image.
+///
+/// A [pending] image is one that has not completed yet. It may also be [weak]ly
+/// tracked, but not [strong].
+///
+/// A [strong] image is being held in the LRU cache, and is subject to eviction
+/// based on [ImageCache.maximumSizeBytes] and [ImageCache.maximumSize]. It may
+/// also be [weak]ly tracked, but not [pending].
+///
+/// A [weak] image is being held until its [ImageStreamCompleter] has no more
+/// listeners. It may also be [pending] or [strong].
+/// An [untracked] image is not being cached.
+///
+/// To obtain an [ImageCacheLocation], use [FlutterImageCache.locationForKey] or
+/// [ImageProvider.findCacheLocation].
+class ImageCacheLocation {
```

Comment:
/cc @jamesderlin who was looking for a way to know if an image was pending in the cache.